File: 백준/Silver/5800. 성적 통계/성적 통계.java
import java.util.*;

public class Main {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int tc = input.nextInt();
		
		for (int i = 1; i <= tc; i++) {
			System.out.println("Class " + i);
			int len = input.nextInt();
			ArrayList<Integer> score = new ArrayList<Integer>();
			for (int j = 0; j < len; j++) {
				score.add(input.nextInt());
			}
			score.sort(Comparator.reverseOrder());
			int mx_gap = 0;
			for (int t = 0; t < score.size() - 1; t++) {
				if (score.get(t) - score.get(t + 1) > mx_gap) {
					mx_gap = score.get(t) - score.get(t + 1);
				}
			}
			System.out.println("Max " + score.get(0) + ", Min " +
			score.get(score.size() - 1) + ", Largest gap " + mx_gap);
		}
	}
}

File: 백준/Silver/5800. 성적 통계/성적 통계.py
import sys
input = sys.stdin.readline


K = int(input())

for tc in range(1, K + 1):
    print(f'Class {tc}')
    score = list(map(int, input().split()))
    p = score.pop(0)
    score.sort(reverse=True)
    max_score = score[0]
    min_score = score[-1]
    mx_gap = 0
    for i in range(1, p):
        if score[i - 1] - score[i] > mx_gap:
            mx_gap = score[i - 1] - score[i]

    print(f'Max {score[0]}, Min {score[-1]}, Largest gap {mx_gap}')